bathbot/src/commands/osu/recent/list.rs

```diff
@@ -649,7 +649,33 @@ async fn process_scores(
                 let a_len = a_map.seconds_drain() as f32 / a.score.mods.clock_rate().unwrap_or(1.0);
                 let b_len = b_map.seconds_drain() as f32 / b.score.mods.clock_rate().unwrap_or(1.0);
 
-                b_len.partial_cmp(&a_len).unwrap_or(Ordering::Equal)
+                let a_completion =
+                    a.score.statistics.total_hits(a.score.mode) as f32 / a_map.n_objects() as f32;
+                let b_completion =
+                    b.score.statistics.total_hits(b.score.mode) as f32 / b_map.n_objects() as f32;
+
+                let a_grade = a.score.grade;
+                let b_grade = b.score.grade;
+
+                let a_is_fail = a_grade.eq_letter(Grade::F);
```

Comment:
`eq_letter` works but is more so intended to view `S` and `SH` as equal; in case of `F` you should just be using `==` instead. Also I think it's fine to use `a.score.grade` directly here without storing it in a variable beforehand.